### PR TITLE
MGMT-3854 log progress report

### PR DESCRIPTION
**Pull request by slaviered**
/hold
/assign @ybettan

### CODE REVIEW

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 1]
```diff
@@ -726,13 +726,17 @@ func (c controller) collectMustGatherLogs(ctx context.Context) (string, error) {
 // We will take logs of assisted controller and upload them to assisted-service
 // by creating tar gz of them.
 func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFunc, wg *sync.WaitGroup, status *ControllerStatus) {
+	progress_ctx := utils.GenerateRequestContext()
```
Comment (eranco74) [inline src/assisted_installer_controller/assisted_installer_controller.go:729]: @tsorya, please review

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 2]
```diff
@@ -726,15 +726,22 @@ func (c controller) collectMustGatherLogs(ctx context.Context) (string, error) {
...
 	podName := ""
 	ticker := time.NewTicker(LogsUploadPeriod)
+	c.log.Infof("Start sending logs")
+	_ = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateRequested)
 	for {
 		select {
 		case <-ctx.Done():
 			if podName != "" {
 				c.log.Infof("Upload final controller and cluster logs before exit")
+				lerr = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateRequested)
```
Comment (tsorya) [inline src/assisted_installer_controller/assisted_installer_controller.go:741]: lets move all this logic to function and just call it here

File: src/installer/installer.go [hunk 3]
```diff
@@ -123,11 +123,18 @@ func (i *installer) InstallNode() error {
 			return err
 		}
 	}
+	//upload host logs and report log status before reboot
+	err = i.inventoryClient.HostLogProgressReport(ctx, i.Config.ClusterID, i.Config.HostID, models.LogsStateRequested)
```
Comment (tsorya) [inline src/installer/installer.go:127]: it is not needed here, you have as part of logsender

Comment (slaviered) [reply src/installer/installer.go:127]: I need it for the case where log sender is stuck or can not communicate with the service: We saw such cases in the triage

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 4]
```diff
@@ -743,6 +750,7 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 					return err == nil
 				})
 			}
+			_ = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateCompleted)
```
Comment (tsorya) [inline src/assisted_installer_controller/assisted_installer_controller.go:753]: so we can stuck with waiting for logs if this api call will fail?

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 5]
```diff
@@ -726,15 +726,22 @@ func (c controller) collectMustGatherLogs(ctx context.Context) (string, error) {
...
+	c.log.Infof("Start sending logs")
+	_ = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateRequested)
 	for {
 		select {
 		case <-ctx.Done():
 			if podName != "" {
 				c.log.Infof("Upload final controller and cluster logs before exit")
+				lerr = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateRequested)
+				if lerr != nil {
+					c.log.WithError(lerr).Errorln("failed to report log progress requested")
```
Comment (tsorya) [inline src/assisted_installer_controller/assisted_installer_controller.go:743]: maybe warning?

File: src/assisted_installer_controller/assisted_installer_controller.go [hunk 6]
```diff
@@ -772,6 +780,11 @@ func (c *controller) UploadLogs(ctx context.Context, cancellog context.CancelFun
 				c.log.WithError(err).Warnf("Failed to upload controller logs")
 				continue
 			}
+			// let the service know that we are still expecting more logs
+			lerr = c.ic.ClusterLogProgressReport(progress_ctx, c.ClusterID, models.LogsStateRequested)
+			if lerr != nil {
+				c.log.WithError(lerr).Errorln("failed to report log progress requested after uploading the logs")
```
Comment (tsorya) [inline src/assisted_installer_controller/assisted_installer_controller.go:786]: error and not Errorln ?

File: src/assisted_installer_controller/assisted_installer_controller.go (hunk 4 above)
Comment (slaviered) [reply src/assisted_installer_controller/assisted_installer_controller.go:753]: There is a timeout for that, so eventually we will stop waiting

File: go.mod [hunk 7]
```diff
@@ -39,6 +39,8 @@ require (
 
 replace (
 	github.com/metal3-io/baremetal-operator => github.com/openshift/baremetal-operator v0.0.0-20200715132148-0f91f62a41fe // Use OpenShift fork
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20200901182017-7ac89ba6b971
```
Comment (tsorya) [inline go.mod:42]: why?

File: src/installer/installer.go [hunk 8]
```diff
@@ -123,11 +123,15 @@ func (i *installer) InstallNode() error {
 			return err
 		}
 	}
+	//upload host logs and report log status before reboot
+	i.inventoryClient.HostLogProgressReport(ctx, i.Config.ClusterID, i.Config.HostID, models.LogsStateRequested)
```
Comment (tsorya) [inline src/installer/installer.go:127]: no need for completed?

File: go.mod (hunk 7 above)
Comment (slaviered) [reply go.mod:42]: because of the bump in assisted-service dependency

File: src/installer/installer.go (hunk 8 above)
Comment (slaviered) [reply src/installer/installer.go:127]: completed is in the logs_sender module

File: go.mod (hunk 7 above)
Comment (slaviered) [reply go.mod:42]: @filanov please review the dependency and comment if necessary

**Pull request merged**